@shared/api/external.ts

```diff
@@ -48,8 +69,16 @@ export const submitTransaction = async (
   return signedTransaction;
 };
 
-export const requestNetwork = async (): Promise<string> => {
-  let response = { network: "", error: "" };
+export const requestNetwork = async (): Promise<NetworkDetails> => {
```

Comment:
this method now returns the full NetworkDetails instead of just the name of the network